Recompiling module `with_type_re':
  type `with_type_re_2.map_pred'/2 was modified.
Recompiling module `with_type_re':
  inst `with_type_re_2.map_pred'/0 was modified.
Recompiling module `with_type_re':
  function `with_type_re_2.with_type_2'/2 was modified.
Recompiling module `with_type_re':
  addition of predicate `with_type_re_2.with_type_3'/4 could cause an ambiguity
  with predicate `with_type_re.with_type_3'/4.
with_type_re.m:019: In clause for predicate `main'/2:
with_type_re.m:019:   error: wrong number of arguments (4; should be 5)
with_type_re.m:019:   in call to predicate `with_type_1'.
